generators/entity-server/templates/src/main/java/package/service/dto/EntityDTO.java.ejs

```diff
@@ -150,34 +148,34 @@ _%>
             return false;
         }
 
-        <%_ if (!embedded) { _%>
+<%_ if (!embedded) { _%>
         <%= dtoClass %> <%= dtoInstance %> = (<%= dtoClass %>) o;
         if (<%- idNames.map(n => `this.${n} == null`).join(' && ') %>){
             return false;
         }
         return <%- idNames.map(n => `Objects.equals(this.${n}, ${dtoInstance}.${n})`).join(' &&\n            ') %>;
-        <%_ } else { _%>
+<%_ } else { _%>
         return false;
-        <%_ } _%>
+<%_ } _%>
     }
 
     @Override
     public int hashCode() {
-    <%_ if (!embedded) { _%>
+<%_ if (!embedded) { _%>
         return Objects.hash(<%- idNames.map(n => `this.${n}`).join(', ') %>);
-    <%_ } else { _%>
+<%_ } else { _%>
         return getClass().hashCode();
-    <%_ } _%>
+<%_ } _%>
     }
 
     // prettier-ignore
     @Override
     public String toString() {
         return "<%= dtoClass %>{" +
-            <%_ for ([idx, reference] of dtoReferences.entries()) {
-                const isNumeric = reference.type && ['integer', 'long', 'float', 'double', 'bigdecimal'].includes(reference.type.toLowerCase()); _%>
-            "<%= idx === 0 ? '' : ', ' %><%= reference.name %>=<% if (! isNumeric) { %>'<% } %>" + <%= buildJavaGet(reference) %> <% if (! isNumeric) { %>+ "'" <% } %>+
-            <%_ } _%>
+<%_ for ([idx, reference] of dtoReferences.entries()) {
+  const isNumeric = (reference.field && reference.field.fieldTypeNumeric) ||  (reference.relationship && reference.relationship.otherEntity.primaryKey && reference.relationship.otherEntity.primaryKey.typeNumeric)_%>
+            "<%= idx === 0 ? '' : ', ' %><%= reference.name %>=<% if (!isNumeric) { %>'<% } %>" + <%= buildJavaGet(reference) %> <% if (! isNumeric) { %>+ "'" <% } %>+
```

Comment:
We can simplify the condition.
```suggestion
  const isQuoted = reference.field && !reference.field.fieldTypeNumeric _%>
            "<%= idx === 0 ? '' : ', ' %><%= reference.name %>=<% if (isQuoted) { %>'<% } %>" + <%= buildJavaGet(reference) %> <% if (!isQuoted) { %>+ "'" <% } %>+
```

Reply:
I really prefer the numeric keyword, which is more meaningful than the quoted one. Also, I tried that 'simplification', but the primaryKey use case was also to be taken into account in the loop.

Reply:
I really prefer the numeric keyword, which is more meaningful than the quoted one. Also, I tried that 'simplification', but the primaryKey use case was also to be taken into account in the loop.

Reply:
Testing the relationship primaryKey or and field it will/can generate something like:
Relationship with User:
- For jwt it will generate `user=User{id=1}`
- For oauth2 it will generate `user='User{id='UUID-...'}'`